File: algorithms/python/lowerBoundSearch.py
from typing import List


def lowerbound(arr: List[int], target: int) -> int:
    left, right = 0, len(arr)

    while left < right:
        mid = (left + right) // 2

        if arr[mid] < target:
            left = mid + 1
        else:
            right = mid

    return right


if __name__ == "__main__":
    n = int(input("> array size: "))
    list_ = []
    for i in range(n):
        list_.append(int(input("> " + str(i + 1) + ": " )))

    list_.sort()
    target = int(input("> target: "))
    print(lowerbound(list_, target))

File: algorithms/cpp/lowerBoundSearch.cpp
#include<algorithm>
#include<iostream>
#include<vector>
using namespace std;


int lowerBound(vector<int>& arr, int target) {
	int left = 0, right = arr.size() - 1;

	while (left < right) {
		int mid = (left + right) / 2;

		if (arr[mid] < target) {
			left = mid + 1;
		}
		else {
			right = mid;
		}
	}

	return right;
}
int main() {
	int N;
	cin >> N;

	vector<int> arr(N);
	for (int i = 0; i < N; i++) {
		cin >> arr[i];
	}
	sort(arr.begin(), arr.end());

	int target;
	cin >> target;

	cout << lowerBound(arr, target);

	return 0;
}